### ISSUE TITLE
Reliability of TimerTrigger

### ISSUE THREAD

**Issue by DiegoFaFe** [Azure/azure-functions-host#5660]
#### Is your question related to a specific version? If so, please specify:
Version 2.0
#### What language does your question apply to? (e.g. C#, JavaScript, Java, All) 
C# with Microsoft.NET.Sdk.Functions-1.0.29
#### Question
Are time triggers reliable? I've a fuction with the following trigger:
[TimerTrigger("00 */30 8-19 * * 1-5")]

So it should be executed every 30m, but as you can see in the following image, it skipped 4 executions:
-9:00
-9:30
-12:30
-13:30

![image](https://user-images.githubusercontent.com/11487214/74748967-ac101400-5269-11ea-86cb-4a8d441ed3a3.png)

**Comment by fabiocav** [comment]
Yes, they should be reliable. If you can share more details about your Function App, like an invocation ID, region and timeframe, we can take a closer look to understand what happened here.

**Comment by AartBluestoke** [comment]
is it possible that the execution took longer than 30 mins? i believe (but i might be wrong) a timer won't trigger if the prior instance is still running.

**Comment by fabiocav** [comment]
@AartBluestoke overlapped executions wouldn't occur, but if you're monitoring the schedule (`UseMonitor`), which is the default for timers with intervals under 1 minute, you'd get a past due invocation with the context appropriately indicating that.

The question is still valid, though, if `useMonitor` is set to false.

**Comment by valdisiljuconoks** [comment]
I'm experiencing similar issue with timers. 
Runtime: v1
Timer: "*/4 * * * * *"
UseMonitor: true

Function.json:

```json
{
  "generatedBy": "Microsoft.NET.Sdk.Functions-1.0.27",
  "configurationSource": "attributes",
  "bindings": [
    {
      "type": "timerTrigger",
      "schedule": "*/4 * * * * *",
      "useMonitor": true,
      "runOnStartup": false,
      "name": "myTimer"
    }
  ],
  "disabled": false,
  "scriptFile": "../bin/Functions.dll",
  "entryPoint": "Functions.DataPumpFunction.Run"
}
```

![image](https://user-images.githubusercontent.com/1767179/79319813-7d34b800-7f11-11ea-93fe-1c70d4a64e88.png)

Workaround for this was to disable & enable function. That restarted trigger and now function is running again.
However, we have real-time critical system and this shows that function was down almost whole night. Which is not good at all for us.

Any hints what and where and how to diagnose? I understand that timers are pretty "interesting" topic to get it right in general in programming ;)

**Comment by balag0** [comment]
@valdisiljuconoks  If you can share the function app name that would be very helpful. 
If not please share the region and os (windows / linux)

**Comment by valdisiljuconoks** [comment]
rthub-collectors-funcapp-prod

**Comment by AsifulNobel** [comment]
Is there any update on this? I am also facing similar issues for a 30 minute interval timer. For example, for the last two weeks, I have seen that there was a period between 6/11/2020 and 6/16/2020 when no timers were triggered. But after 6/17/2020 about 12 PM, I saw timers getting triggered from 6/11/2020 on 6/11/2020. I am sure the Functions App were running smoothly, because the HTTP triggers were working fine.

This type of issue is also present in the last week where one or two interval is missing, where the trigger is not firing at all. I am also seeing instances where different intervals are running close to each other such as 3:30 PM and 11:30 PM timer is triggered within 2 minutes. My timer trigger function is a fire once type of function that starts a Durable Orchestration function without even any retry. It does nothing else. So, I don't see the why such issues are happening.

The Function App is running on Windows OS, US-East location, Runtime Version is 2.0 and on App Service Plan.

**Comment by fengyanglang** [comment]
We are experiencing the same problem. Our timer trigger is scheduled to run at 10 am UTC daily. But it ended up running only 8 times in the past 30 days. The gaps from each runs are about 3 to 6 days without obvious patterns.

The TimerTrigger is set as:
[TimerTrigger("0 10 * * *", RunOnStartup = false, UseMonitor = true)]

**Comment by zmarty** [comment]
+1, we use Azure Functions v3 on Windows - Premium Plan

The machines are NOT under load.

We use with UseMonitor = true and the trigger still skips a lot of runs :(

**Comment by balag0** [comment]
Please include the app names. You can also share the app name privately using https://github.com/Azure/azure-functions-host/wiki/Sharing-Your-Function-App-name-privately

**Comment by fengyanglang** [comment]
Can you use this information to find the app?

Location: East US
Log: Executing '[Azure -Function-Name]' (Reason='Timer fired at 2020-07-08T10:00:00.1835707+00:00', Id=ff5a550b-db47-4b9e-89bd-8e330383f8c0)

**Comment by zmarty** [comment]
Just to add some background info, both @fengyanglang and myself are working in MSFT on an internal project.

**Comment by kroffdaddy** [comment]
It seems to be a known problem, according to [this wiki page](https://github.com/Azure/azure-functions-host/wiki/Investigating-and-reporting-issues-with-timer-triggered-functions-not-firing), that using the "Azure Functions" task in an Azure DevOps release pipeline to deploy a function app _running on a consumption plan_ will not register the timer schedule with the Azure Functions runtime.

It would be great if one would not have to sprinkle magic after deployment to make the timer triggers work.

It's a Function App on Windows on runtime version ~3.
Location: West Europe
Log: 2020-11-10T10:20:00.084 [Information] Executing '<function-name>' (Reason='Timer fired at 2020-11-10T10:20:00.0785874+00:00', Id=39770785-8781-41e7-8e2e-86bc4d661102)
